README.md

```diff
@@ -2,11 +2,26 @@
 
 ## Description
 
-TODO
+This charm creates self hosted github runners. Each unit of this charm will create start a configurable number of runners in LXD containers. Each runner performs only one job
+after which the LXD is destroyed preventing the job from interacting with any other jobs. The charm will periodically check the number of currently active runners and spawn a new
+runner as necessary. Both the interval of the check and the number of runners to maintain are configurable.
 
 ## Usage
 
-TODO
+There are two mandatory configuration options, `path` and `token`. Path is used to determine the organization or repository that the runner will be registered to. This value can
+not be changed, the charm should be redeployed with a new value instead. The token is a github personal access token. This token requires `repository-specific agents: repo` for
+repositories and `organisation-wide agents: repo and admin:org` for use with an organization. This is necessary because the charm will be creating and removing runners in order to
+make runners single use which protects jobs from leaking information to other jobs run on the same runner.
```

Comment:
```suggestion
not be changed, the charm should be redeployed with a new value instead. The token is a [GitHub Personal Access Token (PAT)](https://github.com/settings/tokens) (note: this is not the same as the token given in the Add a Runner instructions). This token requires `repository-specific agents: repo` ("Full control of private repositories" on the website) for
use with repositories and `organisation-wide agents: repo and admin:org` ("Full control of private repositories" and "Full control of orgs and teams, read and write org projects" on the website) for use with an organization. This is necessary because the charm will be creating and removing runners in order to
make runners single use which protects jobs from leaking information to other jobs run on the same runner.
```